### ISSUE TITLE
Nullable Enums do not bind in XAML

### ISSUE THREAD

**Issue by aritchie** [dotnet/maui#24139]
### Description

Nullable enums fail to bind for a bindable property within a custom control.  The setter isn't even hit.  

### Steps to Reproduce

1. Create a control that has a nullable enum as a bindable property
2. Create a page to use control
3. Create a viewmodel
4. Wire up binding and observe

Attached sample is a m

### Link to public reproduction project repository

https://github.com/aritchie/maui_nullable_enum_repro

### Version with bug

8.0.80 SR8

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

All

### Did you find any workaround?

Don't use nullable enums.  Add an unset/none value to the enum.

**Comment by Redth** [comment]
@StephaneDelcroix @simonrozsival  this looks binding related, perhaps a change from more recent trimming/aot work?

**Comment by simonrozsival** [comment]
I don't think this is related to any of the trimming/aot changes because most of those changes went into the net9.0 branch and this is .NET 8.

**Comment by jfversluis** [comment]
Unfortunately not fixed by #28550 and still an issue in the latest .NET 9 version.
